packages/app/src/providers/h5grove/h5grove-api.ts

```diff
@@ -71,6 +73,22 @@ export class H5GroveApi extends ProviderApi {
     return attributes.length > 0 ? this.fetchAttrValues(path) : {};
   }
 
+  public getTiffUrl(
+    dataset: Dataset<ArrayShape, NumericType>,
+    selection: string | undefined
+  ): string | undefined {
+    const { baseURL, params } = this.client.defaults;
+
+    const searchParams = new URLSearchParams(params as Record<string, string>);
+    searchParams.set('path', dataset.path);
+
+    if (selection) {
+      searchParams.set('selection', selection);
+    }
+
+    return `${baseURL as string}/tiff/?${searchParams.toString()}`;
```

Comment:
This is a bit complicated, because it turns out that axios doesn't currently provide a way to retrieve the URL it constructs internally based on the initial configuration (`baseUrl` and `params` in our case)...

Reply:
[axios.getUri](https://github.com/axios/axios#axiosgeturiconfig) does not give what you want ?

Reply:
https://github.com/axios/axios/pull/3737 :cry:

Reply:
Unfortunate. You could still make use of it for the parameters serialization but at the risk of being broken when/if the return value include `baseUrl`.

Let's keep it like this for now :+1: